src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -140,6 +168,10 @@ IFACEMETHODIMP_(void) FancyZones::Destroy() noexcept
         DestroyWindow(m_window);
         m_window = nullptr;
     }
+    if (m_fancyZonesDestroyedEvent) {
+        SetEvent(m_fancyZonesDestroyedEvent.get());
+    }
+    RegCloseKey(m_virtualDesktopsRegKey);

```

Comment:
Better to call `RegCloseKey` only if `RegOpenKeyEx` succeeded. Can you set a default value for `m_virtualDesktopsRegKey` so you can check it before calling `RegCloseKey`?

Reply:
`m_virtualDesktopsRegKey` is handle, so I think `nullptr` can be default